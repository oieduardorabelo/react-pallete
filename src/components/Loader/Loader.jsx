//
// cherry picked from:
// https://codepen.io/jonitrythall/pen/dNJRRK
//
import React from 'react';

import styles from './LoaderStyles';

const animation = ['01', '02', '03', '04', '05', '06', '07', '08', '09','10'];

const Loader = () => (
  <div style={styles.container}>
    <div style={styles.containerSvg}>
      {animation.map(key => (
        <svg key={key} style={styles.svg} height='80' width='210'>
          <ellipse
            cx='25'
            cy='20'
            fill='none'
            rx='10' ry='10'
            style={styles[`ellipse${key}`]}
          />
        </svg>
      ))}
    </div>
  </div>
)

export default Loader;
