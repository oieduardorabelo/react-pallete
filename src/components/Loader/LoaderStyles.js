import { mix } from '../../domain';

const ellipse = {
  transformOrigin: 'center',
}

const LoaderStyles = {
  container: {
    alignItems: 'center',
    animation: 'fadeIn 200ms linear',
    display: 'flex',
    flexDirection: 'column',
    height: '100vh',
    justifyContent: 'center',
    position: 'relative',
    width: '100vw',
  },
  containerSvg: {
    height: 80,
    position: 'relative',
    width: 210,
  },
  svg: {
    position: 'absolute',
  },
  ellipse01: mix(ellipse, {
    animation: 'jump 600ms infinite ease-in-out',
    animationDelay: '0ms',
    cx: '25px',
    opacity: 0.7,
    stroke: '#F1725D',
    strokeWidth: '3px',
  }),
  ellipse02: mix(ellipse, {
    animation: 'jump 600ms infinite ease-in-out',
    animationDelay: '75ms',
    cx: '65px',
    opacity: 0.7,
    stroke: '#38BDAB',
    strokeWidth: '3px',
  }),
  ellipse03: mix(ellipse, {
    animation: 'jump 600ms infinite ease-in-out',
    animationDelay: '150ms',
    cx: '105px',
    opacity: 0.7,
    stroke: '#9D30A5',
    strokeWidth: '3px',
  }),
  ellipse04: mix(ellipse, {
    animation: 'jump 600ms infinite ease-in-out',
    animationDelay: '225ms',
    cx: '145px',
    opacity: 0.7,
    stroke: '#B779E2',
    strokeWidth: '3px',
  }),
  ellipse05: mix(ellipse, {
    animation: 'jump 600ms infinite ease-in-out',
    animationDelay: '300ms',
    cx: '185px',
    opacity: 0.7,
    stroke: '#683893',
    strokeWidth: '3px',
  }),
  ellipse06: mix(ellipse, {
    animation: 'shadow 600ms infinite ease-in-out',
    animationDelay: '0ms',
    cx: '25px',
    cy: '48px',
    fill: '#333333',
    opacity: 0.05,
    rx: 0,
    ry: 0,
  }),
  ellipse07: mix(ellipse, {
    animation: 'shadow 600ms infinite ease-in-out',
    animationDelay: '75ms',
    cx: '65px',
    cy: '48px',
    fill: '#333333',
    opacity: 0.05,
    rx: 0,
    ry: 0,
  }),
  ellipse08: mix(ellipse, {
    animation: 'shadow 600ms infinite ease-in-out',
    animationDelay: '150ms',
    cx: '105px',
    cy: '48px',
    fill: '#333333',
    opacity: 0.05,
    rx: 0,
    ry: 0,
  }),
  ellipse09: mix(ellipse, {
    animation: 'shadow 600ms infinite ease-in-out',
    animationDelay: '225ms',
    cx: '145px',
    cy: '48px',
    fill: '#333333',
    opacity: 0.05,
    rx: 0,
    ry: 0,
  }),
  ellipse10: mix(ellipse, {
    animation: 'shadow 600ms infinite ease-in-out',
    animationDelay: '300ms',
    cx: '185px',
    cy: '48px',
    fill: '#333333',
    opacity: 0.05,
    rx: 0,
    ry: 0,
  }),
};

export default LoaderStyles;
