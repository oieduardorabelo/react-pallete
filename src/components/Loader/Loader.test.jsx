import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Loader from './Loader';

it('renders correctly', () => {
  const wrapper = shallow(
    <Loader />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
