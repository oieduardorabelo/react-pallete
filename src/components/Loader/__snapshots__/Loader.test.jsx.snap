exports[`test renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "animation": "fadeIn 200ms linear",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "justifyContent": "center",
      "position": "relative",
      "width": "100vw",
    }
  }>
  <div
    style={
      Object {
        "height": 80,
        "position": "relative",
        "width": 210,
      }
    }>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "jump 600ms infinite ease-in-out",
            "animationDelay": "0ms",
            "cx": "25px",
            "opacity": 0.7,
            "stroke": "#F1725D",
            "strokeWidth": "3px",
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "jump 600ms infinite ease-in-out",
            "animationDelay": "75ms",
            "cx": "65px",
            "opacity": 0.7,
            "stroke": "#38BDAB",
            "strokeWidth": "3px",
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "jump 600ms infinite ease-in-out",
            "animationDelay": "150ms",
            "cx": "105px",
            "opacity": 0.7,
            "stroke": "#9D30A5",
            "strokeWidth": "3px",
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "jump 600ms infinite ease-in-out",
            "animationDelay": "225ms",
            "cx": "145px",
            "opacity": 0.7,
            "stroke": "#B779E2",
            "strokeWidth": "3px",
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "jump 600ms infinite ease-in-out",
            "animationDelay": "300ms",
            "cx": "185px",
            "opacity": 0.7,
            "stroke": "#683893",
            "strokeWidth": "3px",
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "shadow 600ms infinite ease-in-out",
            "animationDelay": "0ms",
            "cx": "25px",
            "cy": "48px",
            "fill": "#333333",
            "opacity": 0.05,
            "rx": 0,
            "ry": 0,
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "shadow 600ms infinite ease-in-out",
            "animationDelay": "75ms",
            "cx": "65px",
            "cy": "48px",
            "fill": "#333333",
            "opacity": 0.05,
            "rx": 0,
            "ry": 0,
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "shadow 600ms infinite ease-in-out",
            "animationDelay": "150ms",
            "cx": "105px",
            "cy": "48px",
            "fill": "#333333",
            "opacity": 0.05,
            "rx": 0,
            "ry": 0,
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "shadow 600ms infinite ease-in-out",
            "animationDelay": "225ms",
            "cx": "145px",
            "cy": "48px",
            "fill": "#333333",
            "opacity": 0.05,
            "rx": 0,
            "ry": 0,
            "transformOrigin": "center",
          }
        } />
    </svg>
    <svg
      height="80"
      style={
        Object {
          "position": "absolute",
        }
      }
      width="210">
      <ellipse
        cx="25"
        cy="20"
        fill="none"
        rx="10"
        ry="10"
        style={
          Object {
            "animation": "shadow 600ms infinite ease-in-out",
            "animationDelay": "300ms",
            "cx": "185px",
            "cy": "48px",
            "fill": "#333333",
            "opacity": 0.05,
            "rx": 0,
            "ry": 0,
            "transformOrigin": "center",
          }
        } />
    </svg>
  </div>
</div>
`;
